### ISSUE TITLE
Running out of memory (16GB) when rendering EoC Chapter 10 "ExpPolynomial" scene

### ISSUE THREAD

**Issue by bonsairobo** [3b1b/manim#27]
It's all in the title. I imagine you don't experience this issue, and I'm curious how much memory you have. Could this be a memory leak, or is rendering really that memory hungry?

**Comment by danwilhelm** [comment]
Make sure you're using 64-bit Python. With 32-bit Python, your program is limited to 4GB RAM.

**Comment by bonsairobo** [comment]
I watched my used memory crawl from 4 GB to 13 GB while running, so I'm quite certain I'm not running into that limitation.

**Comment by sramsay64** [comment]
I tried this and crashed my computer:

![img_20170506_164940](https://cloud.githubusercontent.com/assets/6789285/25771419/71f7a4f6-328f-11e7-9e1a-4727787beb97.jpg)

I reckon that you could work around this by adding a mighty big chunk of swap space. And if you don't want to dedicate your computer to just running this then make a VM and allocate it some portion of your total ram and inside that VM, add a big chuck of swap and then render it from the VM.

I'd be very interested in seeing how Grant got around this issue.

**Comment by mirefek** [comment]
I also struggle with this issue -- my computer have just 8G of memory and the manim library is really memory-hungry. The way I am attending this issue is that I render scenes by parts in following form:

```python

  def construct()
    self.skip_animations = True
    # some skipped animation
    # ...

    self.skip_animations = False
    # rendered part of the scene
    # ...

    return
    # the remaining code

```

But I have to admit that this is not very convenient.

As I understand it, the scene saves to itself all rendered pictures (to memory). Static scenes are stored just as list of pointers to the same data but large amount of animation causes problems. It seems to me that the problem could be avoided simply by immediate saving the rendered picture to a file instead of storing it into memory. I am planning to investigate the issue later but I hope someone solves it until then :-)

**Comment by styxtwo** [comment]
> As I understand it, the scene saves to itself all rendered pictures (to memory). Static scenes are stored just as list of pointers to the same data but large amount of animation causes problems. It seems to me that the problem could be avoided simply by immediate saving the rendered picture to a file instead of storing it into memory. I am planning to investigate the issue later but I hope someone solves it until then :-)

I wrote an implementation of this a while back (it's in my branch). It slows down rendering so much, I didn't think it was very useful. 

Another idea I had was exporting an mp4 every set amount of frames, and later stitching them together. This will break some global effects (like inverting colours), but I don't think that's a big problem.

**Comment by mirefek** [comment]
@3b1b told me that the issue is already fixed according to my proposal but I have not tested it yet.

**Comment by bonsairobo** [comment]
I believe this issue is solved by commit [18b2267](https://github.com/3b1b/manim/commit/18b2267c7e8744e74ee3df9c63e5460f027522fc)
